build_configs/build_configs.lua

```diff
@@ -0,0 +1,212 @@
+-- define module: build_configs
+local build_configs = build_configs or {}
+
+local gameDefines = { "UE_GAME" }
+
+TARGET_TYPES = {
+    ["Game"] = {
+        ["defines"] = {
+            table.unpack(gameDefines)
+        }
+    },
+    ["CasePreserving"] = {
+        ["defines"] = {
+            "WITH_CASE_PRESERVING_NAME",
+            table.unpack(gameDefines)
+        }
+    }
+}
+
+CONFIG_TYPES = {
+    ["Dev"] = {
+        ["defines"] = {
+            "UE_BUILD_DEVELOPMENT",
+            "STATS"
+        }
+    },
+    ["Debug"] = {
+        ["defines"] = {
+            "UE_BUILD_DEBUG"
+        }
+    },
+    ["Shipping"] = {
+        ["symbols"] = {"debug"},
+        ["defines"] = {
+            "UE_BUILD_SHIPPING"
+        }
+    },
+    ["Test"] = {
+        ["defines"] = {
+            "UE_BUILD_TEST",
+            "STATS"
+        }
+    }
+}
+
+PLATFORM_TYPES = {
+    ["Win64"] = {
+        ["defines"] = {
+            "PLATFORM_WINDOWS",
+            "PLATFORM_MICROSOFT",
+            "OVERRIDE_PLATFORM_HEADER_NAME=Windows",
+            "UBT_COMPILED_PLATFORM=Win64"
+        }
+    }
+}
+
+CLANG_COMPILE_OPTIONS = {
```

Comment:
hi, can you also make this option platform specific? (or allow platform to add/remove flags idk which fits better) e.g. I will add `-fno-delete-null-pointer-checks`  (actually I'm wonder if this is also required on windws if build with clang) and remove `-gcodeview` on linux.  And maybe add `-static-libgcc -static-libstdc++` in link flags.

Reply:
For platform specific compiler flags you could do this:

```
PLATFORM_TYPES = {
    ["Win64"] = {
        ["defines"] = {
            "PLATFORM_WINDOWS",
            "PLATFORM_MICROSOFT",
            "OVERRIDE_PLATFORM_HEADER_NAME=Windows",
            "UBT_COMPILED_PLATFORM=Win64"
        }
    },
    ["Linux"] = {
        ["cxxflags"] = {
            "clang::-fno-delete-null-pointer-checks"
        }
    }
}
```

Every setting for each configuration just gets added to the target using `target:add`, so to learn what else you can add you can look at the target description scope [cxxflags docs](https://xmake.io/#/manual/project_target?id=targetadd_cxxflags)